specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimtagresources.json

```diff
@@ -0,0 +1,186 @@
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Lists a collection of TagResource entities.",
+            "schema": {
+              "$ref": "#/definitions/TagResourceCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/TagResourceContract"
+      }
+    }
+  },
+  "definitions": {
+    "TagResourceCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TagResourceContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Tag list representation."
+    },
+    "TagResourceContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagResourceContractProperties",
+          "description": "TagResource entity contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
+      ],
+      "description": "Contract details."
+    },
+    "TagResourceContractProperties": {
+      "properties": {
+        "tag": {
+          "$ref": "./apimtags.json#/definitions/TagContract",
+          "description": "Tag associated with the resource."
+        },
+        "api": {
+          "$ref": "./apimapis.json#/definitions/ApiContract",
+          "description": "Api associated with the tag."
+        },
+        "operation": {
+          "$ref": "#/definitions/OperationEntityContract",
+          "description": "Api associated with the tag."
+        },
+        "product": {
+          "$ref": "./apimproducts.json#/definitions/ProductContract",
+          "description": "Product associated with the tag."
+        }
+      },
+      "description": "TagResource contract properties."
+    },
+    "OperationEntityContract": {
+      "properties": {
+        "displayName": {
```

Comment:
is there any max/min length to any of the properties in the contract.

Reply:
This is read-only contract, we never use in any put/patch requests. Should we specify min/max length anyway?

Reply:
then we can specify "readonly": "true"

Reply:
Done